ADR/0009-pipeline-service-via-operator.md

```diff
@@ -0,0 +1,74 @@
+# 0009 Pipeline Service via Operator
+
+Created: 2023-09-29
+Last Updated: 2023-09-29
+
+## Status
+
+Proposed
+
+## Context
+
+kcp is no longer being used as a control plane for RHTAP. This means that
+"Pipeline Service" cannot be deployed as an independent service. For our
+initial MVP, all Tekton APIs need to be deployed onto a standard OpenShift
+cluster (specifically OpenShift Dedicated).
+
+## Decision
+
+All Tekton APIs will be provided using the stock OpenShift Pipelines operator (OSP).
+In the spirit of developing in a "Service First" manner, RHTAP will deploy
+a candidate "nightly" release of the operator. The service will be defined in
+the [pipeline-service](https://github.com/openshift/pipeline-service)
+repository, which is then imported into
+[infra-deployments](https://github.com/redhat-appstudio/infra-deployments) as
+an ArgoCD application.
+
+Not all metrics required for operating the service are exposed natively by the
```

Comment:
See https://github.com/tektoncd/experimental/pull/953

Reply:
@vdemeester I was going to add https://github.com/tektoncd/experimental/pull/953 to the huddle agenda for Oct 11 but it looks like we have an all hands meeting at the same time.

but from what I've seen so far, theoretically I could see us transitioning at least some of the metrics we currently maintain in our "exporter" to that if it got promoted out of experimental, but I think there is a fair bit of logistics and roadmapping we need to do.

At least for this iteration of the ADR, are you OK @vdemeester (are are you OK @Roming22 ) if we add a bullet stating we are monitoring the experimental project?

I could any of these following worlds
- we transition everything in exporter to that new operator
- we live with exporter at least for some time
- we live in a world where we want both the operator,  but also the exporter for those metrics we think we need, but get too much pushback
- I'm also curious about how our associated grafana dashboards will get consumed wrt either upstream on downstream, assuming there is agreement that our dashboards would provide value outside of our RHTAP service specific use cases ... which if I had to bet, I bet pipeline users would find them useful.

Reply:
I agree with Gabe - we can keep our metrics exporter for now, but work upstream to bring our findings to the experimental controller.